projects/plugins/jetpack/modules/masterbar/admin-menu/load.php

```diff
@@ -7,10 +7,56 @@
 
 namespace Automattic\Jetpack\Dashboard_Customizations;
 
-if ( jetpack_is_atomic_site() ) {
-	require_once __DIR__ . '/class-atomic-admin-menu.php';
-	Atomic_Admin_Menu::get_instance();
-} elseif ( ! defined( 'IS_WPCOM' ) || ! IS_WPCOM ) {
+/**
+ * Gets the name of the class that customizes the admin menu.
+ *
+ * @return string Class name.
+ */
+function get_admin_menu_class() {
+	// WordPress.com Atomic sites.
+	if ( jetpack_is_atomic_site() ) {
+		require_once __DIR__ . '/class-atomic-admin-menu.php';
+		return __NAMESPACE__ . '\Atomic_Admin_Menu';
+	}
+
+	// WordPress.com Simple sites.
+	if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+		$blog_id = get_current_blog_id();
+
+		// Domain-only sites.
+		$blog_options   = get_blog_option( $blog_id, 'options' );
+		$is_domain_only = ! empty( $blog_options['is_domain_only'] );
+		if ( $is_domain_only ) {
+			require_once __DIR__ . '/class-domain-only-admin-menu.php';
+			return __NAMESPACE__ . '\Domain_Only_Admin_Menu';
+		}
+
+		// P2 sites.
+		if ( \WPForTeams\is_wpforteams_site( $blog_id ) ) {
```

Comment:
Do we need to call the `wpforteams` lib before we use `WPForTeams`?

Reply:
Yup, good point. Updated in 77a6c8c